packages/fast-usdc/src/add-operators.core.js

```diff
@@ -0,0 +1,103 @@
+import { makeTracer } from '@agoric/internal';
+import {
+  CosmosChainInfoShape,
+  DenomDetailShape,
+  DenomShape,
+} from '@agoric/orchestration';
+import { M } from '@endo/patterns';
+import {
+  FastUSDCTermsShape,
+  FeeConfigShape,
+  FeedPolicyShape,
+} from './type-guards.js';
+import { inviteOracles } from './utils/core-eval.js';
+
+/**
+ * @import {Amount, Brand, DepositFacet, Issuer, Payment} from '@agoric/ertp';
+ * @import {TypedPattern} from '@agoric/internal'
+ * @import {Instance, StartParams} from '@agoric/zoe/src/zoeService/utils'
+ * @import {ManifestBundleRef} from '@agoric/deploy-script-support/src/externalTypes.js'
+ * @import {BootstrapManifest} from '@agoric/vats/src/core/lib-boot.js'
+ * @import {LegibleCapData} from './utils/config-marshal.js'
+ * @import {FastUsdcSF} from './fast-usdc.contract.js'
+ * @import {FastUSDCConfig} from './types.js'
+ */
+
+const trace = makeTracer('FUSD-AddOperators', true);
+
+/** @type {TypedPattern<FastUSDCConfig>} */
+export const FastUSDCConfigShape = M.splitRecord({
+  terms: FastUSDCTermsShape,
+  oracles: M.recordOf(M.string(), M.string()),
+  feeConfig: FeeConfigShape,
+  feedPolicy: FeedPolicyShape,
+  chainInfo: M.recordOf(M.string(), CosmosChainInfoShape),
+  assetInfo: M.arrayOf([DenomShape, DenomDetailShape]),
+});
+
+/**
+ * @typedef { PromiseSpaceOf<{
+ *   fastUsdcKit: FastUSDCKit
+ *  }> & {
+ *   instance: PromiseSpaceOf<{ fastUsdc: Instance<FastUsdcSF> }>;
+ *   issuer: PromiseSpaceOf<{ FastLP: Issuer }>;
+ *   brand: PromiseSpaceOf<{ FastLP: Brand }>;
+ * }} FastUSDCCorePowers
+ *
+ * @typedef {StartedInstanceKitWithLabel & {
+ *   creatorFacet: Awaited<ReturnType<FastUsdcSF>>['creatorFacet'];
+ *   privateArgs: StartParams<FastUsdcSF>['privateArgs'];
+ * }} FastUSDCKit
+ */
+
+/**
+ * @throws if oracle smart wallets are not yet provisioned
+ *
+ * @param {BootstrapPowers & FastUSDCCorePowers } powers
+ * @param {{ options: LegibleCapData<FastUSDCConfig> }} config
+ */
+export const addOperators = async (
+  { consume: { namesByAddress, fastUsdcKit } },
+  config,
+) => {
+  trace(addOperators.name);
+
+  const kit = await fastUsdcKit;
+
+  const { creatorFacet } = kit;
+
+  trace(config);
+
+  // @ts-expect-error XXX LegibleCapData typedef
+  const { oracles } = config.options.structure;
+
+  await inviteOracles({ creatorFacet, namesByAddress }, oracles);
+};
+harden(addOperators);
+
+/**
+ * @param {{
+ *   restoreRef: (b: ERef<ManifestBundleRef>) => Promise<Installation>;
+ * }} utils
+ * @param {{
+ *   options: LegibleCapData<FastUSDCConfig>;
```

Comment:
The builder isn't providing a whole `FastUSDCConfig`.

See `@satisfies` vs `@type` earlier.